iothub/service/src/IotHubConnection.cs

```diff
@@ -114,7 +114,7 @@ public async Task<SendingAmqpLink> CreateSendingLinkAsync(string path, TimeSpan
                     session = await _faultTolerantSession.GetOrCreateAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);
                 }
 
-                Uri linkAddress = ConnectionString.BuildLinkAddress(path);
+                Uri linkAddress = Credential.BuildLinkAddress(path);
```

Comment:
I'm going to guess that this would be difficult, and probably not be worth the effort, but is it possible for building the link address to be done somewhere else? This line reads a bit funnily to me. Credentials shouldn't know or care about CBS specific link addresses

Reply:
This may be a moot point if you rename the Credential type though. The type seems to include connection details (hostname) on top of credentials.

Reply:
Maybe IothubConnectionProperties?

Reply:
Will send out these changes in the next PR.